src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/ui/MonitorCacheDumpPanel.java

```diff
@@ -94,11 +94,6 @@ private static void dumpMonitor(PrintStream tty, ObjectMonitor mon, boolean raw)
 
   private void dumpOn(PrintStream tty) {
     Iterator i = ObjectSynchronizer.objectMonitorIterator();
-    if (i == null) {
-      tty.println("This version of HotSpot VM doesn't support monitor cache dump.");
-      tty.println("You need 1.4.0_04, 1.4.1_01 or later versions");
-      return;
-    }
```

Comment:
This code is called when you choose the hsdb "Tools -> Monitor Cache Dump" menu item. Have you tried doing that? I tried running it against a clhsdb process, and currently it doesn't seem to show any monitors. Does it show one after your fix?

Reply:
> This code is called when you choose the hsdb "Tools -> Monitor Cache Dump" menu item. Have you tried doing that? I tried running it against a clhsdb process, and currently it doesn't seem to show any monitors. Does it show one after your fix?

I just tried with your fix and now it shows one monitor:

```
Monitor Cache Dump (not including JVMTI raw monitors):

ObjectMonitor@0x00007f26040010b0
  _header: 0x1
  _object: 0x00000000a0001750, a java/lang/ref/NativeReferenceQueue$Lock
  _owner: null
  _contentions: 0
  _waiters: 1
  _recursions: 0
```